### PR TITLE
feature/production-flag

### LINKED ISSUE #391: Install for production
## Is your feature request related to a problem? Please describe.

Most dependency managers have a `--production` (or similar) flag to only install for production. A normal `install` without that flag install all dependencies (production + development).

Looking at https://pdm.fming.dev/usage/dependency/#install-the-packages-pinned-in-lock-file this is confusing.

## Describe the solution you'd like

I would like it to mimic how other tools does this and believe this tool incorrectly took the approach of pipenv. See https://github.com/pypa/pipenv/issues/3093#issuecomment-432718562 for more reference of tools (the entire thread is interesting).

The reason is that developers are bound to want to install all (production + dev) dependencies in their daily work. A `pdm install`  is thus quicker to write. It also mimics most (all?) other package managers out there.

### PR DESCRIPTION
**Pull request by frostming**
## Pull Request Check List

- [x] A news fragment is added in `news/` describing what is new.
- [x] Test cases added for changed code.

## Describe what you have changed in this PR.

Close #391

### CODE REVIEW

File: docs/docs/usage/dependency.md [hunk 1]
```diff
@@ -136,34 +136,50 @@ To remove existing dependencies from project file and the library directory:
...
+dev1 = ["pytest"]
+dev2 = ["mkdocs"]
+```
+
+| Command                         | What it does                                                             | Comments                  |
+| ------------------------------- | ------------------------------------------------------------------------ | ------------------------- |
+| `pdm install`                   | install prod and dev deps(no optional)                                   |                           |
+| `pdm install -s extra1`         | install prod deps, dev deps, and "extra1" optional section               |                           |
+| `pdm install -s dev1`           | install prod deps and only "dev1" dev section                            |                           |
+| `pdm install -s:all`            | install prod deps and "extra1", "extra2" optional section                |                           |
```
Comment (pawamoy) [inline docs/docs/usage/dependency.md:175]: Shouldn't this one also install all dev dependencies?

File: docs/docs/usage/dependency.md [hunk 2]
```diff
@@ -136,34 +136,50 @@ To remove existing dependencies from project file and the library directory:
...
+extra2 = ["django"]
+
+[tool.pdm.dev-dependencies]  # This is dev dependencies
+dev1 = ["pytest"]
+dev2 = ["mkdocs"]
+```
+
+| Command                         | What it does                                                             | Comments                  |
+| ------------------------------- | ------------------------------------------------------------------------ | ------------------------- |
+| `pdm install`                   | install prod and dev deps(no optional)                                   |                           |
```
Comment (pawamoy) [inline docs/docs/usage/dependency.md:172]:
```suggestion
| `pdm install`                   | install prod and dev deps (no optional)                                  |                           |
```

File: docs/docs/usage/dependency.md (hunk 1 above)
Comment (frostming) [reply docs/docs/usage/dependency.md:175]: Oh year, you are right

File: docs/docs/usage/dependency.md (hunk 2 above)
Comment (frostming) [reply docs/docs/usage/dependency.md]: Fixed in 88d438e

File: docs/docs/usage/dependency.md [hunk 3]
```diff
@@ -136,34 +136,50 @@ To remove existing dependencies from project file and the library directory:
...
+| `pdm install -s extra1`         | install prod deps, dev deps, and "extra1" optional section               |                           |
+| `pdm install -s dev1`           | install prod deps and only "dev1" dev section                            |                           |
+| `pdm install -s:all`            | install prod deps and "extra1", "extra2" optional section                |                           |
+| `pdm install -s extra1 -s dev1` | install prod deps, "extra1" optional section and only "dev1" dev section |                           |
+| `pdm install --prod`            | install prod only                                                        |                           |
+| `pdm install --prod -s extra1`  | installs prod deps and "extra1" optional                                 |                           |
```
[the remark this answers is not in the document]
Comment (frostming) [reply docs/docs/usage/dependency.md]: Fixed in 88d438e

File: docs/docs/usage/dependency.md [hunk 4]
```diff
@@ -136,34 +136,50 @@ To remove existing dependencies from project file and the library directory:
...
+| `pdm install -s extra1`         | install prod deps, dev deps, and "extra1" optional section               |                           |
+| `pdm install -s dev1`           | install prod deps and only "dev1" dev section                            |                           |
+| `pdm install -s:all`            | install prod deps, dev deps and "extra1", "extra2" optional sections     |                           |
+| `pdm install -s extra1 -s dev1` | install prod deps, "extra1" optional section and only "dev1" dev section |                           |
+| `pdm install --prod`            | install prod only                                                        |                           |
+| `pdm install --prod -s extra1`  | install prod deps and "extra1" optional                                  |                           |
+| `pdm install --prod -s dev1`    | Fail, `--prod` can't be given with dev dependencies                      | Leave the `--prod` option |
+
+Dev dependencies are included as long as `--prod` is not passed and `-s` doesn't specify any dev sections.
```
Comment (pawamoy) [inline docs/docs/usage/dependency.md:181]: I'd suggest
```suggestion
**All** development dependencies are included as long as `--prod` is not passed and `-s` doesn't specify any dev sections.
```

**Pull request merged**